### PR TITLE
DOC: Docs for new instrument tests and properties

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #156, #327, and #360. Instrument test upgrade 3 of 3(?).

Updates to documentation including:
- Description of new download flags in `new_instrument.rst`
- Tutorial and explanation of how to configure an instrument for testing
- Updates and style fixes to read the docs

## Type of change

- This change is a documentation update

# How Has This Been Tested?

Using make html

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
Will be writing the tutorials here as the new instrument libraries are created.

**Comment by jklenzing** [comment]
develop-3 => 1246 passed, 15 skipped, 46 warnings in 811.60s (0:13:31)
this branch => 1246 passed, 15 skipped, 46 warnings in 823.32s (0:13:43)

Tests unchanged, as to be expected.

File: docs/supported_instruments.rst [hunk 1]
```diff
@@ -5,31 +5,37 @@ Instrument Templates
...
 
-NASA CDAWeb
------------
+General
+-------
 
-.. automodule:: pysat.instruments.methods.nasa_cdaweb
-   :members: __doc__, init, load, list_files, list_remote_files, download
+.. automodule:: pysat.instruments.methods.general
+   :members: __doc__, list_files,
```
Comment (aburrell) [inline docs/supported_instruments.rst:31]: extra comma

File: pysat/tests/test_instruments.py [hunk 2]
```diff
@@ -31,6 +38,10 @@ class TestInstruments(InstTestClass):
 
     def setup(self):
         """Runs before every method to create a clean testing setup."""
+        # Developers for instrument libraries should update the following line
+        # to point to their own library package
+        # e.g.,
+        # self.package = mypackage.instruments
```
Comment (aburrell) [inline pysat/tests/test_instruments.py:44]: Is there a reason this sentence is broken up into three lines?

**Comment by jklenzing** [comment]
Cleaned up based on comments.  Thanks!

**Pull request merged**